packages/opentelemetry-plugin-document-load/src/documentLoad.ts

```diff
@@ -0,0 +1,283 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BasePlugin, otperformance } from '@opentelemetry/core';
+import { PluginConfig, Span, SpanOptions } from '@opentelemetry/types';
+import { AttributeNames } from './enums/AttributeNames';
+import { PerformanceTimingNames } from './enums/PerformanceTimingNames';
+import { PerformanceEntries, PerformanceLegacy } from './types';
+import { hasKey } from './utils';
+
+// import { IPerformanceEntries, PerformanceEntries, PerformanceLegacy } from './types';
+
+/**
+ * This class represents a document load plugin
+ */
+export class DocumentLoad extends BasePlugin<unknown> {
+  readonly component: string = 'document-load';
+  readonly version: string = '1';
+  moduleName = this.component;
+  protected _config!: PluginConfig;
+
+  /**
+   *
+   * @param {DocumentLoadConfig} config
+   */
+  constructor(config: PluginConfig = {}) {
+    super();
+    this._onDocumentLoaded = this._onDocumentLoaded.bind(this);
+    this._config = config;
+  }
+
+  /**
+   * callback to be executed when page is loaded
+   * @private
+   */
+  private _onDocumentLoaded() {
+    this._collectPerformance();
+  }
+
+  /**
+   * Collects information about performance and creates appropriate spans
+   * @private
+   */
+  private _collectPerformance() {
+    const entries = this._getEntries();
+
+    const rootSpan = this._startSpan(
+      AttributeNames.DOCUMENT_LOAD,
+      PerformanceTimingNames.FETCH_START,
+      entries
+    );
+
+    this._startAndFinishSpan(
+      AttributeNames.DOMAIN_LOOKUP,
+      PerformanceTimingNames.DOMAIN_LOOKUP_START,
+      PerformanceTimingNames.DOMAIN_LOOKUP_END,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // Opening Connection
+    const connectSpan = this._startSpan(
+      AttributeNames.CONNECT,
+      PerformanceTimingNames.CONNECT_START,
+      entries,
+      {
+        parent: rootSpan,
+      }
+    );
+
+    // TLS negotiation
+    if (hasKey(entries, PerformanceTimingNames.SECURE_CONNECTION_START)) {
```

Comment:
What value does the `hasKey` check do here and in the other places, since you are doing a `typeof` check anyway on the value? Do those checks allow us to remove the `hasKey` function entirely?